lib/streamlit/state/safe_session_state.py

```diff
@@ -13,11 +13,21 @@
 # limitations under the License.
 
 import threading
-from typing import Any, Dict, Tuple, Optional, List, Set
+from typing import Any, Dict, Optional, List, Set, Union
+
+import attr
+from typing_extensions import Literal
 
 from streamlit.proto.WidgetStates_pb2 import WidgetState as WidgetStateProto
 from streamlit.proto.WidgetStates_pb2 import WidgetStates as WidgetStatesProto
-from .session_state import SessionState, WidgetMetadata
+
+from .session_state import SessionState, WidgetMetadata, WidgetStateReport, T
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class DisconnectedReport:
+    value: None = attr.ib(default=None, init=False)
+    change: Literal[False] = attr.ib(default=False, init=False)
```

Comment:
This class feels a little strange to me - why does it need to exist? Does it need the `value` and `change` properties? It feels like a detail that `safe_session_state` callers shouldn't need to know about, but I might be misunderstanding!

Reply:
This mirrors `WidgetStateReport`, but is what you get if you're not connected to the state store. The separate class was supposed to signal that this isn't actually a `WidgetStateReport`, because we don't know what the state of the widget is in this case.

But see discussion above about just replacing this with `None`.

Reply:
Realising that what I am trying to do here is essentially a [tagged union](https://en.wikipedia.org/wiki/Tagged_union).

Reply:
I gave a shot at summarizing my thoughts in the PR description.